package com.github.teamfossilsarcheology.fossil.client.renderer.entity;

import com.github.teamfossilsarcheology.fossil.client.model.FriendlyPiglinModel;
import com.github.teamfossilsarcheology.fossil.client.renderer.RendererFabricFix;
import com.github.teamfossilsarcheology.fossil.entity.monster.FriendlyPiglin;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.client.renderer.entity.HumanoidMobRenderer;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.phys.Vec3;
import org.jetbrains.annotations.NotNull;

public class FriendlyPiglinRenderer extends HumanoidMobRenderer<FriendlyPiglin, FriendlyPiglinModel> implements RendererFabricFix {
    public static final ResourceLocation TEXTURE = ResourceLocation.parse("textures/entity/piglin/zombified_piglin.png");

    public FriendlyPiglinRenderer(EntityRendererProvider.Context context) {
        super(context, new FriendlyPiglinModel(), 0.5f);
    }

    @Override
    public @NotNull Vec3 getRenderOffset(FriendlyPiglin entity, float partialTicks) {
        if (entity.isInSittingPose()) {
            return new Vec3(0.0, entity.isBaby() ? -0.2 : -0.4, 0.0);
        }
        return super.getRenderOffset(entity, partialTicks);
    }

    @Override
    public @NotNull ResourceLocation getTextureLocation(FriendlyPiglin entity) {
        return TEXTURE;
    }

    @Override
    public ResourceLocation _getTextureLocation(Entity entity) {
        return getTextureLocation((FriendlyPiglin) entity);
    }
}
